src/lib/common/sol-power-supply-impl-linux.c

```diff
@@ -119,7 +119,9 @@ _get_string_prop(const char *name, enum sol_power_supply_prop prop,
     if (r < 0)
         return r;
 
-    *value = str_value;
+    r = sol_buffer_append_slice(buf, sol_str_slice_from_str(str_value));
```

Comment:
isn't `sol_util_read_file()` returning the length? So you could use it here instead of using that helper that will do the strlen() for you.

Also, eventually provide a  read-file-to-buffer, if we don't have it (I guess we do). Then you read directly to the buffer and avoid the malloc.